krcal/core/io_functions.py

```diff
@@ -50,3 +52,96 @@ def compute_and_save_hist_as_pd(values     : np.array           ,
+                                 title      : str                ,
+                                 x_label    : str                ,
+                                 y_range    : Tuple[float, float],
+                                 norm       : bool = False       )->None:
+    """
+    Computes 1d-histogram and saves it as a pdf image.
+    Parameters
+    ----------
+    values : np.array
+        Array with values to be plotted.
+    out_file: string
+        File where histogram will be saved.
+    n_bins: int
+        Number of bins to make the histogram.
+    range_hist: length-2 tuple (optional)
+        Range of the histogram.
+    title: str
+        Title for the plot.
+    x_label: str
+        Label for X-axis.
+    y_range: length-2 tuple
+        Limit fot Y-axis.
+    norm: bool
+        If True, histogram will be normalized.
+    """
+    fig = plt.figure();
+    plt.hist(values, bins = n_bins,
+             range = range_hist, density = norm,
+             histtype='step', linewidth=2);
+    plt.ylabel('Entries');
+    plt.xlabel(x_label);
+    plt.title(title);
+    plt.ylim(y_range);
+    plt.grid(True, alpha=0.5, color='k', linestyle=':');
+    fig.savefig(out_file.format(title).replace(" ", ""), bbox_inches='tight')
+
+    return
+
+
+def plot_and_save_evolution_figure(time        : np.array   ,
+                                   param_name  : str        ,
+                                   param       : np.array   ,
+                                   param_u     : np.array   ,
+                                   units       : str        ,
+                                   file_name   : str        ,
+                                   n_sigmas_lim: int     = 1):
+    """
+    Plot and save evolution of a given parameter vs time.
+    Parameters
+    ----------
+    time : np.array
+        Array with the centers of each time interval (X-values).
+    param_name : string
+        Name of the parameter to plot.
+    param : np.array
+        Array with the magnitude to plot (Y-values).
+    param_u : np.array
+        Array with uncertainty values.
+    units : string
+        Units to be shown in x-label.
+
+    Returns
+    ----------
+    Nothing
+    """
+    fig = plt.figure()
+    plt.errorbar(time, param, param_u, fmt='.')
+    plt.xlabel('Date')
+    plt.ylabel('{0} ({1})'.format(param_name, units));
+    plt.title('{0}'.format(param_name))
+    mean = np.mean(param)
+    std  = np.std(param)
+    plt.ylim(param.min()-std*n_sigmas_lim,
+             param.max()+std*n_sigmas_lim)
+    plt.grid(True, alpha=0.5, color='k', linestyle=':')
+    ax = plt.gca()
+    xfmt = md.DateFormatter('%d-%m %H:%M')
+    ax.xaxis.set_major_formatter(xfmt)
+    plt.xticks( rotation=25 )
+    plt.text(0.03,0.9,
+             'Mean= {0} \n Std= {1}'.format(np.round(mean,4), np.round(std,4)),
+             fontsize=10, transform=ax.transAxes,
+             bbox={'facecolor': 'white', 'alpha': 1, 'pad': 5})
```

Comment:
Spacing also inconsistent. Not sure, but maybe @gonzaponte can shed light on the style we follow, but I guess that:
```
when_in_line(we=don't, add=spaces)
when_not_in_line(we  = do
                 add = spaces)
```

Reply:
Indeed. But in this case they are combined, so I would give preference to ` = ` over `=`. Maybe even splitting into 1 keyword argument per line.

May I also suggest the use of f-strings and string-formatting syntax?
```
f'Mean= {mean:.4f} \n Std= {std:.4f}'
```

Reply:
Ok, thanks. Regarding your second suggestion we have a question: in the event that you have a variable called, for example: name='Run{0}.pdf', if you want to substitute {0} for a certain value, could it be possible to do it in your way? or only: name.format(runnumber) using format?

Reply:
In the case it is already stored in a variable you need to do `.format`. f-strings are evaluated immediately, so you either do
- `name = f"Run{run}.pdf"` or
- `name = "Run{0}.pdf"; name.format(run)`

Reply:
Ok! thank you.